tensorflow/core/graph/graph_constructor.cc

```diff
@@ -840,7 +840,7 @@ Status GraphConstructor::Convert() {
         back_edges_.push_back(
             EdgeInfo(inputs[i].name, inputs[i].index, node, i));
       } else if (inputs[i].index == Graph::kControlSlot) {
-        g_->AddControlEdge(inputs[i].node, node);
+        g_->AddControlEdge(inputs[i].node, node, /*update_node_def=*/false);
```

Comment:
Do you need to update this file? Also unit tests for your new method?

Reply:
These were just changes you had in your early draft implementation and I copied them over. I'll remove them and check that everything works fine. Also, I'll add tests for the new method once I get the chance this week. :)